profiling-ffi/src/profiles/interning_api.rs

```diff
@@ -0,0 +1,404 @@
+}
+
+/// This functions interns its argument into the profiler.
+/// If successful, it an opaque interning ID.
+/// This ID is valid for use on this profiler, until the profiler is reset.
+/// It is an error to use this id after the profiler has been reset, or on a different profiler.
+/// On error, it holds an error message in the error variant.
+///
+/// # Safety
+/// The `profile` ptr must point to a valid Profile object created by this
+/// module.
+/// All other arguments must remain valid for the length of this call.
+/// This call is _NOT_ thread-safe.
+#[must_use]
+#[no_mangle]
+#[named]
+pub unsafe extern "C" fn ddog_prof_Profile_intern_managed_strings(
+    profile: *mut Profile,
+    strings: Slice<ManagedStringId>,
+    mut out: MutSlice<GenerationalId<StringId>>,
+) -> VoidResult {
+    wrap_with_void_ffi_result!({
+        anyhow::ensure!(strings.len() == out.len());
+        profile_ptr_to_inner(profile)?
+            .intern_managed_strings(strings.as_slice(), out.as_mut_slice())?;
+    })
+}
+
+/// This functions interns its argument into the profiler.
+/// If successful, it an opaque interning ID.
+/// This ID is valid for use on this profiler, until the profiler is reset.
+/// It is an error to use this id after the profiler has been reset, or on a different profiler.
+/// On error, it holds an error message in the error variant.
+///
+/// # Safety
+/// The `profile` ptr must point to a valid Profile object created by this
+/// module.
+/// All other arguments must remain valid for the length of this call.
+/// This call is _NOT_ thread-safe.
+#[must_use]
+#[no_mangle]
+#[named]
+pub unsafe extern "C" fn ddog_prof_Profile_intern_mapping(
+    profile: *mut Profile,
+    memory_start: u64,
+    memory_limit: u64,
+    file_offset: u64,
+    filename: GenerationalId<StringId>,
+    build_id: GenerationalId<StringId>,
+) -> Result<GenerationalId<MappingId>> {
+    wrap_with_ffi_result!({
+        profile_ptr_to_inner(profile)?.intern_mapping(
+            memory_start,
+            memory_limit,
+            file_offset,
+            filename,
+            build_id,
+        )
+    })
+}
+
+/// This functions interns its argument into the profiler.
+/// If successful, it an opaque interning ID.
+/// This ID is valid for use on this profiler, until the profiler is reset.
+/// It is an error to use this id after the profiler has been reset, or on a different profiler.
+/// On error, it holds an error message in the error variant.
+///
+/// # Safety
+/// The `profile` ptr must point to a valid Profile object created by this
+/// module.
+/// All other arguments must remain valid for the length of this call.
+/// This call is _NOT_ thread-safe.
+#[must_use]
+#[no_mangle]
+#[named]
+pub unsafe extern "C" fn ddog_prof_Profile_intern_sample(
+    profile: *mut Profile,
+    stacktrace: GenerationalId<StackTraceId>,
+    values: Slice<i64>,
+    labels: GenerationalId<LabelSetId>,
+    timestamp: Option<NonZeroI64>,
+) -> VoidResult {
```

Comment:
"If successful, it an opaque interning ID." -> doesn't apply to this one, which returns `VoidResult` (also needs a bit of a text pass -- shows up in other places ;D )

Reply:
fixed